executor/common_linux.h

```diff
@@ -1557,6 +1557,80 @@ static long syz_io_uring_submit(volatile long a0, volatile long a1, volatile lon
 
 #endif
 
+#if SYZ_EXECUTOR || __NR_syz_usbip_server_init
+
+#include <errno.h>
+#include <fcntl.h>
+#include <linux/usb/ch9.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+// This should be coherent with CONFIG_USBIP_VHCI_HC_PORTS.
+#define VHCI_HC_PORTS 8
+#define VHCI_PORTS (VHCI_HC_PORTS * 2)
+
+static int usb2_port_num = 0;
+static int usb3_port_num = 0;
+
+static long syz_usbip_server_init(volatile long a0)
+{
+	int fd;
+	int rc;
+	int socket_pair[2];
+	char buffer[100];
+	int server_fd, client_fd;
+	char path[100];
+	int port_num;
+	int available_port_num;
+
+	int speed = (int)a0;
+	bool usb3 = (speed == USB_SPEED_SUPER);
+
+	rc = socketpair(AF_UNIX, SOCK_STREAM, 0, socket_pair);
+	if (rc < 0)
+		fail("syz_usbip_server_init : socketpair failed: %d\n", rc);
+
+	client_fd = socket_pair[0];
+	server_fd = socket_pair[1];
+
+	available_port_num = usb3 ? __atomic_fetch_add(&usb3_port_num, 1, __ATOMIC_RELAXED) : __atomic_fetch_add(&usb2_port_num, 1, __ATOMIC_RELAXED);
```

Comment:
A bit shorter:
`__atomic_fetch_add(usb3 ? &usb3_port_num : &usb2_port_num, 1, __ATOMIC_RELAXED)`

Also it's better to declare usb2/3_port_num right here as they are not used anywhere else.
And maybe even:
```
static int port_alloc[2];
int port = __atomic_fetch_add(&port_alloc[usb3], 1, __ATOMIC_RELAXED) ;
```